cards4-template.js

```diff
@@ -4,8 +4,16 @@ var makeCard = // receive factory with external name `makeCard`
     // The factory itself:
     function makeCard(id) {  //makeCard is also IIFE's internal name
```

Comment:
This is working well, and your code is easy to read, good work! 

When you have time, go back and work on the isValid function, as this is a vital part of your factory. Currently your factory accepts any sort of argument given to it, so that calling makeCard('string') is a perfectly acceptable thing to do. Even though the methods on that particular object won't work as the user may want, you still don't want to end up with an array of cards filled with unacceptable objects.